In [2]:
import pandas as pd
import matplotlib.pyplot as plt
from scipy.stats import linregress, t, f
import numpy as np


1:

In [3]:
num_features = 3

reg_df = num_features # number of features
reg_sum_sq = 2216338
reg_mean_sq = reg_sum_sq / reg_df

res_df = 62
res_sum_sq = 8913083
res_mean_sq = res_sum_sq / res_df

F_val = reg_mean_sq / res_mean_sq

P_F = 1 - f.cdf(F_val, reg_df, res_df)
P_F

0.0030778764811709713

In [14]:
res_mean_sq

143759.40322580645

In [12]:
n = reg_df + res_df + 1
print('n:', n)
print()
df = n - num_features - 1

intercept_est = 3316.4
intercept_std_error = 937.7
intercept_t = intercept_est / intercept_std_error
P_t_intercept = 2 - 2*t.cdf(intercept_t, df)
print('intercept:')
print('t_value:', intercept_t)
print('PR(>|t|):', P_t_intercept)
print()

education_est = 850
education_t_value = 3.646
education_std_error = education_est / education_t_value
P_t_education = 2 - 2*t.cdf(education_t_value, df)
print('education:')
print('std_error:', education_std_error)
print('PR(>|t|):', P_t_education)
print()

experience_est = 932.4
experience_std_error = 260.1
experience_t_value = experience_est / experience_std_error
P_t_education = 2 - 2*t.cdf(experience_t_value, df)
print('education:')
print('t_value:', intercept_t)
print('PR(>|t|):', P_t_intercept)
print()

STEM_t_value = 1.675
STEM_std_error = 330.1
STEM_est = STEM_t_value * STEM_std_error
P_t_STEM = 2 - 2*t.cdf(STEM_t_value, df)
print('STEM:')
print('est:', STEM_est)
print('PR(>|t|):', P_t_STEM)


n: 66

intercept:
t_value: 3.5367388290498027
PR(>|t|): 0.0007733214228629048

education:
std_error: 233.13219967087218
PR(>|t|): 0.0005468345553012455

education:
t_value: 3.5367388290498027
PR(>|t|): 0.0007733214228629048

STEM:
est: 552.9175
PR(>|t|): 0.09897146444217686


2:

$ P\_val = 0.0030778764811709713 < 0.05 $ so we reject that there is no connection

3:

In [13]:
p_val = 1 - t.cdf(experience_t_value, df)
p_val

0.00033225192760155675

so we reject that there is no positive relation

4 + 5:

we have:  
$ {y - a^{T}\hat{\beta} \over \hat{\sigma}\sqrt{a^T(X^TX)^{-1}a+1}} \sim t_{n-p}$

so confidence inteval will be in the form:  

$a^t\hat{\beta} \pm t_{n-p, 1-{\alpha \over 2}}\hat{\sigma}\sqrt{a^T(X^TX)^{-1}a+1}$  
where:  
$\hat{\sigma} = {MSE} = 143759.40$  
$ p = \#features + 1 = 4 $  
$n = 66$  
$\hat{\beta} = (3316.4, 850, 932.4, 552.9)^T $

4: $ a^T = (1, 10, 5, 1)$  
5: $ a^T = (1, 10, 6, 0)$